File: src/main/java/ru/markn/webchat/servicies/UserService.java
package ru.markn.webchat.servicies;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import ru.markn.webchat.models.User;
import ru.markn.webchat.repository.UserRepository;
import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UserService implements IUserService
{
    private final UserRepository userRepository;

    @Override
    public void deleteUser(long id)
    {
        userRepository.deleteById(id);
    }


    @Override
    public List<User> getAllUsers()
    {
        return userRepository.findAll();
    }

    @Override
    public User saveUser(User user)
    {
        return userRepository.save(user);
    }

    @Override
    public User getUserById(long id)
    {
        return userRepository.findById(id).orElseThrow();
    }

    @Override
    public User updateUser(User user)
    {
        Optional<User> existingUser = userRepository.findById(user.getId());
        if (existingUser.isPresent())
        {
            return userRepository.save(user);
        } else
        {
            return null;
        }
    }

    @Override
    public User getUser(long id)
    {
        return userRepository.getReferenceById(id);
    }
}

File: src/main/kotlin/ru/markn/webchat/servicies/UserService.kt
package ru.markn.webchat.servicies

import org.springframework.security.core.userdetails.UserDetailsService
import ru.markn.webchat.dtos.UserSaveDto
import ru.markn.webchat.dtos.UserDto
import ru.markn.webchat.models.User

interface UserService : UserDetailsService {
    val users: List<User>
    fun getUserById(id: Long): User
    fun getUsersById(ids: List<Long>): List<User>
    fun getUserByUsername(username: String): User
    fun getUsersByUsernameIn(usernames: List<String>): List<User>
    fun getUserByEmail(email: String): User
    fun addUser(userDto: UserSaveDto): User
    fun updateUser(userDto: UserDto): User
    fun deleteUser(id: Long)
}